src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,503 @@ title: Builder.io & Astro
+    - **title:** 'First post, woohoo!'
+    - **slug:** 'first-post-woohoo'
+4. Design your post. Click on the **Add Block** button and add something - maybe a text field with some text to start with.
+5. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
+6. When you're ready click the **Publish** button in the upper right corner.
+
+Your content is now live in the Builder CMS but not part of your Astro project yet. You'll be fetching and displaying it in the following steps.
+
+### Displaying a list of blog posts
+
+Create the page `src/pages/index.astro` that will display a list of post titles, each linking to its own page:
+
+<FileTree title="Project Structure">
+- src/
+  - pages/
+    - **index.astro**
+  - env.d.ts
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+Then add the following content:
+
+```astro title="src/pages/index.astro" /title|slug/
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+```
+
+Fetching via the content API returns an array of objects containing data for each post. The `fields` query parameter tells Builder which data is included (see highlighted code). `slug` and `title` should match the names of the custom data fields you've added to [your Builder model](#creating-a-model-for-a-blog-post).
+
+
+Use the `posts` array returned from the fetch to display a list of blog posts on the page:
+
+```astro title="src/pages/index.astro" ins={18-99}
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+
+<html lang="en">
+  <head>
+    <title>Blog Index</title>
+  </head>
+  <body>
+    <ul>
+      {
+        posts.map(({ data: { slug, title } }) => (
+          <li>
+            <a href={`/posts/${slug}`}>{title}</a>
+          </li>
+        ))
+      }
+    </ul>
+  </body>
+</html>
+
+```
+
+Go to your index route and you should be able to see a list of links each with the title of a blog post!
+
+If you encounter any errors, you can always fall back to [Builder's API explorer](https://builder.io/api-explorer).
```

Comment:
```suggestion
```
I don't *love* setting up the expectation of problems.  :sweat_smile:  Let's see how we feel without this!

Reply:
How about reframing it so it doesn't anticipate a problem, but points out a useful resource? 😁 

'Builder's [API-explorer](...) can help with fine-tuning your results.'

Could be an aside, too.